CLAUDE.md

```diff
@@ -101,15 +114,31 @@ The project includes Flox environment configuration (`.flox/env/manifest.toml`)
 - Docker
 - Claude Code CLI
 
-Activate with: `flox activate` (if Flox is installed)
+**ADL CLI Installation:**
+- ADL CLI v0.18.2 is automatically installed when running `task generate:docs` or `task generate`
+- Installation command: `curl -fsSL https://raw.githubusercontent.com/inference-gateway/adl-cli/main/install.sh | bash`
+- Installs to `~/.local/bin/adl` and verifies PATH availability
+
+Activate Flox with: `flox activate` (if Flox is installed)
 
 ## Important Constraints
 
 - **Generated Files**: Never manually edit files with "DO NOT EDIT" headers
-- **Configuration Changes**: Always modify `agent.yaml` and regenerate
-- **ADL Version**: Ensure ADL CLI 0.18.2 or compatible version for regeneration
+- **Configuration Changes**: Always modify `agent.yaml` and regenerate with `task generate`
+- **ADL CLI Requirement**: ADL CLI is automatically installed when running `task generate`
+- **Skills Validation**: All skills in `agent.yaml` must have `id`, `name`, `type`, `tags`, and `schema` fields
+- **Generate Command Success**: The `task generate` command must complete successfully before development can proceed
 - **Port Configuration**: Default 8080, configurable via `A2A_PORT` or `A2A_SERVER_PORT`
 
+## Known Issues
+
+### ADL CLI Template Error
+The `task generate` command currently has a known template error:
+- **Error**: `failed to execute template ai/agents.md: template: template:70:15: executing "template" at <.Type>: can't evaluate field Type`
+- **Impact**: Most files generate successfully, but generation fails at the final template step
+- **Workaround**: The core functionality works - skill validation passes and main files are generated correctly
+- **Status**: This appears to be a bug in ADL CLI v0.18.2 template system rather than configuration issue
+
```

Comment:
```suggestion
```

Fixing it right now on the ADL CLI: https://github.com/inference-gateway/adl-cli/issues/59